### DISCUSSION TITLE
How to use the well-known physical constants?

### DISCUSSION THREAD

**Discussion by eigenein** [iliekturtles/uom#411] [Q&A] [answered]
👋 `uom` seems to have defined a bunch of the well-known constants, such as [`speed_of_light_in_vacuum`](https://docs.rs/uom/latest/uom/si/velocity/struct.speed_of_light_in_vacuum.html) or [`planck_constant`](https://docs.rs/uom/latest/uom/si/action/struct.planck_constant.html).

But I fail to understand how to use them: they don't implement `Quantity`, but instead, they implement `Conversion`. Am I supposed to write something like `Velocity::new::<speed_of_light_in_vacuum>(1.0)` every time I need a constant? Or do I overlook something?

**Comment by iliekturtles** [comment, marked as answer]
Your links to the speed of light and the planck constant are for a context where these are used as units. e.g. `1.0 speed of light in a vacuum` as a unit of velocity is an equivalent way of writing `299_792_458.0 meters per second`.

#32 exists to add universal constants to `uom`. As a work-around you can manually define them in your code: https://github.com/iliekturtles/uom/issues/26#issuecomment-527881570. This also applies for anyone coming to this discussion wondering how to write their own constants.
